Update workspace delete command docs to reference RUM vs empty state (#31999)
hashicorp/terraform: website/docs/cli/commands/workspace/delete.mdx

This command will delete the specified workspace.

> To delete a workspace, it must already exist, it must not have resources under management,
> and it must not be your current workspace. If the workspace has resources under management,

Reviewer: It turns out that even without considering Terraform Cloud workspaces, this is more correct because the requirement is that the state be empty of any resources (not outputs or metadata)

However, Terraform CLI seems to prefer the word "tracked" to "managed" today. I wonder if we should mirror that term here?

```suggestion
To delete a workspace, it must already exist, it must not be tracking resources,
and it must not be your current workspace. If the workspace is tracking resources,
```

![Screen Shot 2022-11-22 at 12 47 16 PM](https://user-images.githubusercontent.com/174332/203407463-50a34626-c5a3-4c8a-b92b-2a739aef8406.png)

Reviewer 2: ah, this is an older draft! @JarrettSpiker product management let me know we're not saying RUM anymore, but more just "managed resources".  So I support removing!

Terraform will not allow you to delete it unless the `-force` flag is specified.

> Workspaces with resources under management (RUM) are workspaces which manage resources that physically exist.

Reviewer: Should we introduce this acronym? Is it a term we want to familiar users with to the extend that we want to use it in the docs? It seems like we usually spell out words like Terraform Cloud instead of using acronyms like TFC

Author: good feedback! For a while there it seemed like we were going to make RUM a thing, but since we are not I have updated this to talk about tracked resources instead, as per your other comment

This means that the workspace has a non-empty state.
Additionally, different [backends](/language/settings/backends/configuration#backend-types) may implement other
restrictions on whether a workspace is considered to have RUM, such as whether the workspace is locked.

If you delete a workspace with resources under management (via `-force`), then resources
may become "dangling". These are resources that physically exist but that
Terraform can no longer manage. This is sometimes preferred: you may want
Terraform to stop managing resources, so they can be managed some other way.
Most of the time, however, this is not intended and so Terraform protects you
from getting into this situation.

The command-line flags are all optional. The only supported flags are:

> * `-force` - Delete the workspace even if it has resources under management. Defaults to false.

Reviewer 2: ```suggestion
* `-force` - Delete the workspace even if it has resources under management. After deletion, Terraform can no longer track or manage the workspace's infrastructure. Defaults to false.
```
